card/lib/card/cache.rb

```diff
@@ -36,41 +36,44 @@ def renew
             raise "renewing nil cache: #{klass}"
           end
         end
-        reset_local
-      end
-
-      def prefix_root
-        @@prefix_root ||= ( cfg = Cardio.config and cfg = cfg.database_configuration and
-                            cfg[Rails.env]['database'] )
+        reset_soft
       end
 
       def system_prefix klass
-        "#{ prefix_root }/#{ klass }"
+        "#{Rails.env}/#{klass}"
       end
 
-      def restore klass=nil
-        reset_local
+      def restore
+        reset_soft
         prepopulate
       end
 
-      def generate_cache_id
-        ((Time.now.to_f * 100).to_i).to_s +
-          ('a'..'z').to_a[rand(26)] +
-          ('a'..'z').to_a[rand(26)]
+      def reset_global # deprecated
+        reset_all
       end
 
-      def reset_global
-        cache_by_class.each do |_klass, cache|
-          cache.reset hard=true
-        end
+      def reset_all
+        reset_hard
+        reset_soft
+
         Card::Codename.reset_cache
         Cardio.delete_tmp_files
       end
 
-      def reset_local
-        cache_by_class.each do |cc, cache|
-          if Card::Cache === cache
-            cache.reset_local
+      def reset_hard
+        cache_by_class.each do |_klass, cache|
+          if cache.is_a? Card::Cache
+            cache.hard.reset
+          else
+            warn "reset class #{cc}, #{cache.class} #{caller[0..8] * "\n"} ???"
```

Comment:
do you ever see these warnings?  we might be able to get rid of this test.  (ancient debugging?)